### PR TITLE
Update `get-codeowners.ps1` to work with the updated `RetrieveCodeOwners` executable + add some tests; make assorted refactorings

### PR DESCRIPTION
**Pull request by konrad-jamrozik**
This PR is part of work required in preparation to merge:
- #5088

Specifically, to enable review of ownership changes due to upcoming wildcards support, as explained in:
- https://github.com/Azure/azure-sdk-tools/pull/5088#issuecomment-1376542202

As such, this PR contributes to:
- #2770

This PR is a prerequisite for following PRs:
- https://github.com/Azure/azure-sdk-tools/pull/5431

### Merging prerequisite

This PR can be merged only after the following PRs are merged and relevant NuGet package published
- https://github.com/Azure/azure-sdk-tools/pull/5437
- #5104

This is because this PR depends on that NuGet package.

### CODE REVIEW

File: eng/common/scripts/get-codeowners.ps1 [hunk 1]
```diff
@@ -80,16 +141,16 @@ function TestGetCodeOwner([string]$targetDirectory, [string]$codeOwnerFileLocati
   }
 }
 
-if($Test) {
+if ($Test) {
```
Comment (weshaggard) [inline eng/common/scripts/get-codeowners.ps1:144]: Now that we are establishing a testing pattern we should probably follow that and move these tests outs into their own tests script. However we can do that as a follow-up if you like.

File: eng/common/scripts/get-codeowners.ps1 [hunk 2]
```diff
@@ -52,7 +113,7 @@ function Get-CodeOwners ([string]$targetDirectory, [string]$codeOwnerFileLocatio
   
   $codeOwnersJson = $codeOwnersString | ConvertFrom-Json
```
Comment (konrad-jamrozik) [inline eng/common/scripts/get-codeowners.ps1:114]: Note to self: need to handle here case when the STDOUT has warnings.

File: eng/common/scripts/get-codeowners.ps1 (hunk 1 above)
Comment (konrad-jamrozik) [reply eng/common/scripts/get-codeowners.ps1:144]: I created an issue to track it:
- https://github.com/Azure/azure-sdk-tools/issues/5434

File: eng/common/scripts/get-codeowners.ps1 [hunk 3]
```diff
@@ -80,21 +159,19 @@ function TestGetCodeOwner([string]$targetDirectory, [string]$codeOwnerFileLocati
...
-  # TestGetCodeOwner -targetDirectory "sdk/noPath" -codeOwnerFileLocation $testFile -includeNonUserAliases $true -expectReturn @("person1", "person2")
-  # TestGetCodeOwner -targetDirectory "/sdk/azconfig" -codeOwnerFileLocation $testFile -includeNonUserAliases $true -expectReturn @("person3", "person4")
-  # TestGetCodeOwner -targetDirectory "/sdk/azconfig/package" -codeOwnerFileLocation $testFile -includeNonUserAliases $true  $testFile -expectReturn @("person3", "person4")
-  # TestGetCodeOwner -targetDirectory "/sd" -codeOwnerFileLocation $testFile -includeNonUserAliases $true  -expectReturn @()
-  # TestGetCodeOwner -targetDirectory "/sdk/testUser/" -codeOwnerFileLocation $testFile -expectReturn @("azure-sdk") 
+  Write-Host "Running reduced test suite at `$PSScriptRoot of $PSSCriptRoot. Please see https://github.com/Azure/azure-sdk-tools/issues/5434 for more."
+
+  $azSdkToolsCodeowners = (Resolve-Path "$PSScriptRoot/../../../.github/CODEOWNERS")
+  TestGetCodeowners -targetPath "eng/common/scripts/get-codeowners.ps1" -codeownersFileLocation $azSdkToolsCodeowners -includeNonUserAliases $true -expectReturn @("sima-zhu", "weshaggard", "benbp")
```
Comment (weshaggard) [inline eng/common/scripts/get-codeowners.ps1:169]: @konrad-jamrozik could you please remove sima-zhu from the CODEOWNERS file as part of this test and update to yourself.

Comment (konrad-jamrozik) [reply eng/common/scripts/get-codeowners.ps1]: done!

**Review by weshaggard — APPROVED** [review]

**Pull request merged**